cmd/kubeops/internal/handler/dashboardcompat.go

```diff
@@ -0,0 +1,121 @@
+// The Dashboard doesn't understand the Helm 3 release format.
+// This file is a compatibility layer that translates Helm 3 releases to a Helm 2-similar format suitable for the Dashboard.
+
+package handler
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/any"
+	"gopkg.in/yaml.v2"
+	h3chart "helm.sh/helm/v3/pkg/chart"
+	h3 "helm.sh/helm/v3/pkg/release"
+	h2chart "k8s.io/helm/pkg/proto/hapi/chart"
+	h2 "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+// generatedYamlHeader is prepended to YAML generated from the internal map[string]interface{} representation.
+const generatedYamlHeader = "# Not original YAML! Generated from parsed representation."
+
+type dashboardCompatibleRelease struct {
+	Name      string                    `json:"name,omitempty"`
+	Info      h2.Info                   `json:"info,omitempty"`
+	Chart     dashboardCompatibleChart  `json:"chart,omitempty"`
+	Config    dashboardCompatibleConfig `json:"config,omitempty"`
+	Manifest  string                    `json:"manifest,omitempty"`
+	Version   int                       `json:"version,omitempty"`
+	Namespace string                    `json:"namespace,omitempty"`
+}
+
+type dashboardCompatibleChart struct {
+	Files     []*any.Any                `json:"files,omitempty"`
+	Metadata  h3chart.Metadata          `json:"metadata,omitempty"`
+	Templates []*h2chart.Template       `json:"templates,omitempty"`
+	Values    dashboardCompatibleValues `json:"values,omitempty"`
+}
+
+type dashboardCompatibleValues struct {
```

Comment:
...and here https://github.com/helm/helm/blob/release-2.16/pkg/proto/hapi/chart/config.pb.go#L22

(in the helm2 code, the `Values` and `Config` appear to be the same thing, which looks reflected in you having the same type).

Reply:
AFAIK, `Values` and `Config` are not the same thing: `Values` is the original `values.yaml` for the chart, whereas `Config` is the _effective_ set of values used for the _release_ – usually `Values` overridden with any custom values set by the user at install time.

Reply:
Sorry - yes, I meant that they appear to use the same type in helm2 rather than requiring two different but identical types - so `Chart.Values` (at https://github.com/helm/helm/blob/release-2.16/pkg/proto/hapi/chart/chart.pb.go#L32) is actually of type `*Config` (whereas here we had two identical types with `dashboardCompatibleValues` and `dashboardCompatibleConfig` I think?)